### PR TITLE
release: 4.0.1

### LINKED ISSUE #4675: release: 4.0.1
## Checklist

- [x] Assign to next release manager
- [x] Close previous release milestone
- [ ] Create next release milestone
- [ ] Associate issue with next release milestone

### PR DESCRIPTION
**Pull request by jsjoeio**
This PR is to generate a new release of `code-server` at `4.0.1`

## Screenshot

https://user-images.githubusercontent.com/3806031/148128070-43f0f16c-5c81-4ea8-b35b-1595d79b5b03.mov

## TODOs

Follow "Publishing a release" steps in `ci/README.md`

- [x] rebase after merge https://github.com/coder/code-server/pull/4680
- [x] bump helm chart version like https://github.com/coder/code-server/pull/4273/files
- [x] document helm chart version (should be automated) in an issue: https://github.com/coder/code-server/issues/4682
- [x] update vscode version after merge https://github.com/coder/vscode/pull/37
- [x] update `CHANGELOG`
- [x] check that built-in extensions work on actual build
- [x] fix typo in CHANGELOG and release notes @jsjoeio
- [x] address regression https://github.com/coder/code-server/issues/3782#issuecomment-1005715493 @jsjoeio 
- [x] address issue https://github.com/coder/code-server/issues/1510#issuecomment-1005712247 @code-asher 
- [x] publish release and merge PR
- [x] update the AUR package

Fixes #4675 

## Notes
Found this issue and documented: https://github.com/coder/code-server/issues/4679

### CODE REVIEW

File: ci/build/release-prep.sh [hunk 1]
```diff
@@ -83,7 +83,7 @@ main() {
   echo -e "Great! We'll prep a PR for updating to $CODE_SERVER_VERSION_TO_UPDATE\n"
   $CMD rg -g '!yarn.lock' -g '!*.svg' -g '!CHANGELOG.md' --files-with-matches --fixed-strings "${CODE_SERVER_CURRENT_VERSION}" | $CMD xargs sd "$CODE_SERVER_CURRENT_VERSION" "$CODE_SERVER_VERSION_TO_UPDATE"
 
-  $CMD git commit -am "chore(release): bump version to $CODE_SERVER_VERSION_TO_UPDATE"
+  $CMD git commit --no-verify -am "chore(release): bump version to $CODE_SERVER_VERSION_TO_UPDATE"
```
Comment (jsjoeio) [inline ci/build/release-prep.sh:86]: 🧹 Could be my machine but when I run this script, Husky/git precommit fails. Adding `--no-verify` here so there are no issues running the script.

File: ci/helm-chart/Chart.yaml [hunk 2]
```diff
@@ -15,9 +15,9 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 1.0.5
+version: 2.0.0
```
Comment (jsjoeio) [inline ci/helm-chart/Chart.yaml:18]: ⬆️ Bumping major versions since we bumped the major version of code-server

**Review by code-asher — APPROVED** [review]

**Comment by benz0li** [comment]
Please fix https://github.com/coder/code-server/issues/3782#issuecomment-1005566987 before releasing.

There is also https://github.com/coder/code-server/issues/1510#issuecomment-1005462384 which is in contradiction to https://github.com/coder/code-server/issues/1510#issuecomment-861632752.

**Comment by jsjoeio** [comment]
Tested again locally and all looks good!

## Video

https://user-images.githubusercontent.com/3806031/148281538-4bd772bc-79ef-40a1-b242-9f9103f93668.mov

**Pull request merged**